src/Controller/CreatePayPalOrderFromPaymentPageAction.php

```diff
@@ -20,28 +22,33 @@
 
 final class CreatePayPalOrderFromPaymentPageAction
 {
-    private FactoryInterface $stateMachineFactory;
-
-    private ObjectManager $paymentManager;
-
-    private PaymentStateManagerInterface $paymentStateManager;
-
-    private OrderProviderInterface $orderProvider;
-
-    private CapturePaymentResolverInterface $capturePaymentResolver;
-
     public function __construct(
-        FactoryInterface $stateMachineFactory,
-        ObjectManager $paymentManager,
-        PaymentStateManagerInterface $paymentStateManager,
-        OrderProviderInterface $orderProvider,
-        CapturePaymentResolverInterface $capturePaymentResolver
+        private readonly FactoryInterface|StateMachineInterface $stateMachineFactory,
+        private readonly ObjectManager $paymentManager,
+        private readonly PaymentStateManagerInterface $paymentStateManager,
+        private readonly OrderProviderInterface $orderProvider,
+        private readonly CapturePaymentResolverInterface $capturePaymentResolver,
     ) {
-        $this->stateMachineFactory = $stateMachineFactory;
-        $this->paymentManager = $paymentManager;
-        $this->paymentStateManager = $paymentStateManager;
-        $this->orderProvider = $orderProvider;
-        $this->capturePaymentResolver = $capturePaymentResolver;
+        if ($this->stateMachineFactory instanceof FactoryInterface) {
+            trigger_deprecation(
+                'sylius/paypal-plugin',
+                '1.6',
+                sprintf(
+                    'Passing an instance of "%s" as the first argument is deprecated and will be prohibited in 2.0. Use "%s" instead.',
+                    FactoryInterface::class,
+                    StateMachineInterface::class,
+                ),
+            );
+        }
+
+        trigger_deprecation(
+            'sylius/paypal-plugin',
+            '1.6',
+            sprintf(
+                'Passing an instance of "%s" as the second argument is deprecated and will be prohibited in 2.0',
+                ObjectManager::class,
+            ),
+        );
```

Comment:
We should make the `$paymentManager` nullable and set a null in the service, then put this deprecation in a condition so we won't spam the end users.

Reply:
Fixed